### PR TITLE
feat: Port `ImageAutomationPeer`

### PR DESCRIPTION
**Pull request by morning4coffe-dev**
GitHub Issue (If applicable): part of #14344 

## PR Type

What kind of change does this PR introduce?
- Feature

## What is the new behavior?
Ported the `ImageAutomationPeer` from WinUI source

## PR Checklist

Please check if your PR fulfills the following requirements:

- [ ] Docs have been added/updated which fit [documentation template](https://github.com/unoplatform/uno/blob/master/doc/.feature-template.md) (for bug fixes / features)
- [ ] [Unit Tests and/or UI Tests](https://github.com/unoplatform/uno/blob/master/doc/articles/uno-development/working-with-the-samples-apps.md) for the changes have been added (for bug fixes / features) (if applicable)
- [ ] Validated PR `Screenshots Compare Test Run` results.
- [x] Contains **NO** breaking changes
- [x] Associated with an issue (GitHub or internal) and uses the [automatic close keywords](https://help.github.com/en/github/managing-your-work-on-github/linking-a-pull-request-to-an-issue).
- [x] Commits must be following the [Conventional Commits](https://www.conventionalcommits.org/en/v1.0.0/#summary) specification.

### CODE REVIEW

File: src/Uno.UI/UI/Xaml/Automation/Peers/ImageAutomationPeer.cs [hunk 1]
```diff
@@ -1,19 +1,48 @@
...
+	protected override string GetFullDescriptionCore()
+	{
+		var length = base.GetFullDescriptionCore()?.Length ?? 0;
+
+		if (length == 0)
 		{
-			return AutomationControlType.Image;
+			var image = Owner as Controls.Image;
+
+			//TODO UNO Description on Item is not implemented
```
Comment (MartinZikmund) [inline src/Uno.UI/UI/Xaml/Automation/Peers/ImageAutomationPeer.cs:42]: On `Image` not `Item` I assume 👍

**Review by MartinZikmund — APPROVED** [review]

File: src/Uno.UI/UI/Xaml/Automation/Peers/ImageAutomationPeer.cs (hunk 1 above)
Comment (MartinZikmund) [reply src/Uno.UI/UI/Xaml/Automation/Peers/ImageAutomationPeer.cs:42]: I would suggest just creating an internal `Description` property in `Image` that always returns empty string and move the comment there instead.

Comment (morning4coffe-dev) [reply src/Uno.UI/UI/Xaml/Automation/Peers/ImageAutomationPeer.cs]: Sounds good, thanks, Martin!

**Review by ajpinedam — APPROVED** [review]

File: src/Uno.UI/UI/Xaml/Controls/Image/Image.shared.cs [hunk 2]
```diff
@@ -26,4 +26,13 @@ partial class Image
 			OnSourceChanged(Source, forceReload: true);
 		}
 	}
+
+	/// <summary>
+	/// Returns an Empty string as the Description for the Image.
+	/// </summary>
+	internal string Description
```
Comment (MartinZikmund) [inline src/Uno.UI/UI/Xaml/Controls/Image/Image.shared.cs:33]: You have to add `new` keyword, as this exists in `NSObject`

File: src/Uno.UI/UI/Xaml/Controls/Image/Image.shared.cs [hunk 3]
```diff
@@ -26,4 +26,13 @@ partial class Image
 			OnSourceChanged(Source, forceReload: true);
 		}
 	}
+
+	/// <summary>
+	/// Returns an Empty string as the Description for the Image.
+	/// </summary>
+	internal new string Description
```
Comment (MartinZikmund) [inline src/Uno.UI/UI/Xaml/Controls/Image/Image.shared.cs:33]: The `new` must be in `#if __MACOS__`

Comment (morning4coffe-dev) [reply src/Uno.UI/UI/Xaml/Controls/Image/Image.shared.cs]: Aaa, I see. Thanks! 🌞

**Pull request merged**